vision/google/cloud/vision/image.py

```diff
@@ -30,22 +30,28 @@ class Image(object):
     :type content: bytes
     :param content: Byte stream of an image.
 
+    :type filename: str
+    :param filename: Filename to image.
+
     :type source_uri: str
     :param source_uri: Google Cloud Storage URI of image.
 
     :type client: :class:`~google.cloud.vision.client.Client`
     :param client: Instance of Vision client.
     """
 
-    def __init__(self, client, content=None, source_uri=None):
+    def __init__(self, client, content=None, filename=None, source_uri=None):
```

Comment:
You need an explicit check that makes sure exact one of `content`, `filename` and `source_uri` is not None. The first thing that comes to mind is:

```python
(int(content is not None) + int(filename is not None) + 
 int(source_uri is not None)) == 1
# OR FEWER CHARS
(int(content is None) + int(filename is None) + 
 int(source_uri is None)) == 2
```

though it feels kind of icky.

Maybe @jonparrott or @tseaver can make me feel better about writing code like that / suggest something less icky.